radix-engine/src/updates/protocol_updates.rs

```diff
@@ -0,0 +1,102 @@
+use super::state_updates::*;
+use crate::{internal_prelude::*, track::StateUpdates};
+use radix_substate_store_interface::interface::SubstateDatabase;
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
+pub enum ProtocolUpdate {
+    /// Introduces BLS12-381 and Keccak-256 features.
+    Bls12381AndKeccak256,
+
+    /// Exposes second-precision timestamp.
+    SecondPrecisionTimestamp,
+
+    /// Increases the math precision with native pool implementations.
+    PoolMathPrecisionFix,
+
+    /// Changes the cost associated with validator creation.
+    ValidatorCreationFeeFix,
+
+    /// Exposes a getter method for reading owner role rule.
+    OwnerRoleGetter,
+
+    /// Various system patches.
+    SystemPatches,
+
+    /// Introduces the account locker blueprint.
+    AccountLocker,
+
+    /// Moves various protocol parameters to state.
+    ProtocolParamsToState,
+}
+
+pub struct ProtocolUpdates {
+    updates: BTreeSet<ProtocolUpdate>,
+}
+
+impl ProtocolUpdates {
+    pub fn none() -> Self {
+        Self {
+            updates: BTreeSet::new(),
+        }
+    }
+
+    pub fn all() -> Self {
+        Self::none().with_anemone().with_bottlenose()
+    }
+
+    pub fn with_anemone(mut self) -> Self {
+        self.updates.extend(btreeset!(
+            ProtocolUpdate::Bls12381AndKeccak256,
+            ProtocolUpdate::SecondPrecisionTimestamp,
+            ProtocolUpdate::PoolMathPrecisionFix,
+            ProtocolUpdate::ValidatorCreationFeeFix,
+        ));
+        self
+    }
+
+    pub fn with_bottlenose(mut self) -> Self {
```

Comment:
Not sure but maybe `with_bottlenose()` should include `with_anemone()`? 🤔 
If not then maybe worth clearly stating that given named update does not include previous updates (however it might happen that some updates depend on previous updates).

Reply:
Fair enough :)